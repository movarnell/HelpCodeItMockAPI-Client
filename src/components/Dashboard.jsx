import React from 'react';
import { Link } from 'react-router-dom';
//import { IoArrowBack } from 'react-icons/io5';
import Navigation from './Navigation';

function Dashboard() {
  return (
    <div className="min-h-screen bg-gray-100">
      <Navigation />
      <div className="py-6 mx-auto max-w-7xl sm:px-6 lg:px-8">
        <div className="flex items-center mb-6">
          {/* <Link
            to="/"
            className="mr-4 text-indigo-600 transition-colors duration-200 hover:text-indigo-800"
          >
            <IoArrowBack className="w-6 h-6" />
          </Link> */}
          <h1 className="text-3xl font-bold text-gray-900">Dashboard</h1>
            <div className="mx-auto space-x-4">
              <h3 className="text-2xl font-semibold text-gray-800 ms-4">Latest Updates</h3>
              <ul className="list-disc list-inside">
                <li> New and improved UI</li>
                <li>

                  Added ability to add data using JSON input
                </li>
              </ul>
            </div>
        </div>
        <div className="px-4 py-6 sm:px-0">
          <div className="flex flex-col items-center justify-center space-y-6 border-4 border-gray-200 border-dashed rounded-lg h-96">
            <h2 className="mb-4 text-3xl font-semibold text-gray-800">Welcome to Your Dashboard</h2>
            <div className="flex space-x-4">
              <Link
                to="/endpoints/create"
                className="px-6 py-3 font-medium text-white transition duration-150 ease-in-out bg-blue-600 rounded-md hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
              >
                Create New Endpoint
              </Link>
              <Link
                to="/endpoints"
                className="px-6 py-3 font-medium text-white transition duration-150 ease-in-out bg-green-600 rounded-md hover:bg-green-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500"
              >
                View Endpoints
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Dashboard;