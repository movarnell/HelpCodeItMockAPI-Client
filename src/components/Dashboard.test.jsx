import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer';

// Mock the Navigation component
jest.mock('./Navigation', () => () => <div data-testid="navigation" />);

describe('Dashboard Component', () => {
  const renderComponent = () => {
    render(
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>
    );
  };

  test('renders dashboard title', () => {
    renderComponent();
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
  });

  test('renders create endpoint button', () => {
    renderComponent();
    expect(screen.getByText('Create New Endpoint')).toBeInTheDocument();
  });

  test('renders view endpoints button', () => {
    renderComponent();
    expect(screen.getByText('View Endpoints')).toBeInTheDocument();
  });

  test('renders navigation component', () => {
    renderComponent();
    expect(screen.getByTestId('navigation')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});