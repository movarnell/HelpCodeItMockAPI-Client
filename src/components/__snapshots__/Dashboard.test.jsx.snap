// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component matches snapshot 1`] = `
<div
  className="min-h-screen bg-gray-100"
>
  <div
    data-testid="navigation"
  />
  <div
    className="py-6 mx-auto max-w-7xl sm:px-6 lg:px-8"
  >
    <div
      className="flex items-center mb-6"
    >
      <h1
        className="text-3xl font-bold text-gray-900"
      >
        Dashboard
      </h1>
      <div
        className="mx-auto space-x-4"
      >
        <h3
          className="text-2xl font-semibold text-gray-800 ms-4"
        >
          Latest Updates
        </h3>
        <ul
          className="list-disc list-inside"
        >
          <li>
             New and improved UI
          </li>
          <li>
            Added ability to add data using JSON input
          </li>
        </ul>
      </div>
    </div>
    <div
      className="flex flex-col items-center justify-center p-6 space-y-4 border-4 border-gray-200 border-dashed rounded-lg"
    >
      <div
        className="w-full max-w-xs"
      >
        <img
          alt="logo"
          className="object-contain w-full h-auto"
          src="test-file-stub"
        />
      </div>
      <h2
        className="text-2xl font-semibold text-center text-gray-800"
      >
        Welcome to Your Dashboard
      </h2>
      <div
        className="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-4"
      >
        <a
          className="px-4 py-2 font-medium text-white transition duration-150 ease-in-out bg-blue-600 rounded-md hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
          href="/endpoints/create"
          onClick={[Function]}
        >
          Create New Endpoint
        </a>
        <a
          className="px-4 py-2 font-medium text-white transition duration-150 ease-in-out bg-green-600 rounded-md hover:bg-green-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500"
          href="/endpoints"
          onClick={[Function]}
        >
          View Endpoints
        </a>
      </div>
    </div>
  </div>
</div>
`;
